### PR TITLE
update dependencies.

### PR DESCRIPTION
**Pull request by Medesis**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,11 +2,11 @@ sudo: false
 language: ruby
 cache: bundler
 rvm:
-  - 2.1
-  - 2.2
+  - 2.3.4
```
Comment (websi) [inline .travis.yml]: Please use `2.2`, `2.3` and `2.4`. 

See https://github.com/dkdeploy/dkdeploy-core/commit/82ff875a1821aa2e189d25bfeac0c9def6534b60#diff-354f30a63fb0907d4ad57269548329e3

Comment (ncreuschling) [inline .travis.yml]: We should add Ruby `2.2` here as well. Use `2.3` and `2.4`. (See `dkdeploy-core`)

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,15 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
 - `Dkdeploy::Php::Helpers::Http::http_get_with_redirect` gets arguments as options hash now
+- rubocop upgrade to 0.50
+- ruby upgrade to 2.2
+- bundler upgrade
+- rake upgrade
+- dkdeploy-test_environment upgrade to 2.0
+- dkdeploy-core upgrade 9.0
+- Add dependencies for cucumber, rubocop, aruba and mysql2
+
+
```
Comment (websi) [inline CHANGELOG.md]: Please remove empty lines

File: dkdeploy-php.gemspec [hunk 3]
```diff
@@ -12,17 +11,20 @@ Gem::Specification.new do |spec|
...
   spec.require_paths = ['lib']
 
-  spec.add_development_dependency 'bundler', '~> 1.12.5'
-  spec.add_development_dependency 'rake', '~> 11.2'
-  spec.add_development_dependency 'rubocop', '0.46'
-  spec.add_development_dependency 'dkdeploy-test_environment', '~> 1.0'
+  spec.add_development_dependency 'bundler'
+  spec.add_development_dependency 'rake'
+  spec.add_development_dependency 'cucumber', '~> 2.4'
+  spec.add_development_dependency 'rubocop', '~> 0.50'
```
Comment (websi) [inline dkdeploy-php.gemspec]: Please use `'~> 0.50.0'`

Comment (ncreuschling) [inline dkdeploy-php.gemspec]: Make it consistent with dkdeploy-core: `~> 0.50.0`

File: lib/dkdeploy/php/tasks/composer.rake [hunk 4]
```diff
@@ -24,7 +24,7 @@ namespace :composer do
       begin
         # run composer status
         invoke 'composer:local:run', :status
-      rescue
+      rescue ComposerError
```
Comment (ncreuschling) [inline lib/dkdeploy/php/tasks/composer.rake]: Where does this exception class come from?

**Review by ncreuschling — CHANGES_REQUESTED** [review]

File: lib/dkdeploy/php/tasks/composer.rake (hunk 4 above)
Comment (websi) [inline lib/dkdeploy/php/tasks/composer.rake]: Please use `SSHKit::StandardError`

File: spec/fixtures/application/config/deploy/dev.rb [hunk 5]
```diff
@@ -1,5 +1,5 @@
 set :deploy_to, '/var/www/dkdeploy'
-server 'dkdeploy-php.dev', roles: %w(web app backend), primary: true
+server 'dkdeploy-php.dev', roles: %w[web app backend], primary: true
```
Comment (websi) [inline spec/fixtures/application/config/deploy/dev.rb:2]: `web`, `app` and `backend` are strings or symbols.

File: .travis.yml [hunk 6]
```diff
@@ -2,11 +2,11 @@ sudo: false
...
+  - 2.4.1
 
 before_install:
-  - gem install bundler --version 1.12.5 --no-document
+  - gem install bundler --no-document
 
 script:
-  - bundle exec rubocop
+  - bundle exec rubocop --display-cop-names --extra-details
```
Comment (ncreuschling) [inline .travis.yml]: Please import these settings into `.rubocop.yml`. (See `dkdeploy-core`)

File: spec/fixtures/application/config/deploy/dev.rb [hunk 7]
```diff
@@ -18,11 +18,11 @@
 end
 
 set :copy_source, 'htdocs'
-set :copy_exclude, %w(
+set :copy_exclude, %w[
```
Comment (websi) [inline spec/fixtures/application/config/deploy/dev.rb:21]: The content of the array are `strings`. No variables or methods.

**Review by ncreuschling — CHANGES_REQUESTED** [review]

File: spec/fixtures/capistrano/configuration/server_with_faulty_domain_configuration.rb [hunk 8]
```diff
@@ -1,2 +1,2 @@
 
-server 'dkdeploy-php.dev', roles: %w(web app backend), domain: 'not-existing-dkdeploy.dev'
+server 'dkdeploy-php.dev', roles: %w[web app backend], domain: 'not-existing-dkdeploy.dev'
```
Comment (websi) [inline spec/fixtures/capistrano/configuration/server_with_faulty_domain_configuration.rb:2]: The content of the array are `strings`. No variables or methods.

File: spec/fixtures/capistrano/configuration/server_with_ssl.rb [hunk 9]
```diff
@@ -1 +1 @@
-server 'dkdeploy-php.dev', roles: %w(web app backend), primary: true, domain_scheme: 'https'
+server 'dkdeploy-php.dev', roles: %w[web app backend], primary: true, domain_scheme: 'https'
```
Comment (websi) [inline spec/fixtures/capistrano/configuration/server_with_ssl.rb:1]: The content of the array are `strings`. No variables or methods.

File: spec/fixtures/capistrano/configuration/server_with_valid_domain.rb [hunk 10]
```diff
@@ -1,2 +1,2 @@
 
-server 'dkdeploy-php.dev', roles: %w(web app backend), domain: 'second-dkdeploy-php.dev'
+server 'dkdeploy-php.dev', roles: %w[web app backend], domain: 'second-dkdeploy-php.dev'
```
Comment (websi) [inline spec/fixtures/capistrano/configuration/server_with_valid_domain.rb:2]: The content of the array are `strings`. No variables or methods.

**Review by websi — CHANGES_REQUESTED** [review]

File: spec/fixtures/application/config/deploy/dev.rb (hunk 5 above)
Comment (websi) [reply spec/fixtures/application/config/deploy/dev.rb:2]: My fault. No changes required.

File: spec/fixtures/application/config/deploy/dev.rb (hunk 7 above)
Comment (websi) [reply spec/fixtures/application/config/deploy/dev.rb:21]: My fault. No changes required.

File: spec/fixtures/capistrano/configuration/server_with_faulty_domain_configuration.rb (hunk 8 above)
Comment (websi) [reply spec/fixtures/capistrano/configuration/server_with_faulty_domain_configuration.rb:2]: My fault. No changes required.

File: spec/fixtures/capistrano/configuration/server_with_ssl.rb (hunk 9 above)
Comment (websi) [reply spec/fixtures/capistrano/configuration/server_with_ssl.rb:1]: My fault. No changes required.

File: spec/fixtures/capistrano/configuration/server_with_valid_domain.rb (hunk 10 above)
Comment (websi) [reply spec/fixtures/capistrano/configuration/server_with_valid_domain.rb:2]: My fault. No changes required.

File: .rubocop.yml [hunk 11]
```diff
@@ -22,3 +22,7 @@ Metrics/AbcSize:
   Max: 30
 Style/SignalException:
   EnforcedStyle: only_raise
+Style/FormatStringToken:
+  Enabled: false
+Metrics/BlockLength:
+  Enabled: false
```
Comment (ncreuschling) [inline .rubocop.yml]: We should use the same options as in `dkdeploy-core`:

```
  TargetRubyVersion: 2.2
  ExtraDetails: true
  DisplayStyleGuide: true
  DisplayCopNames: true
```

Comment (ncreuschling) [inline .rubocop.yml:31]: We should use the same options as in `dkdeploy-core`:

```
  TargetRubyVersion: 2.2
  ExtraDetails: true
  DisplayStyleGuide: true
  DisplayCopNames: true
```

**Review by ncreuschling — CHANGES_REQUESTED** [review]

**Review by ncreuschling — APPROVED** [review]

**Comment by ncreuschling** [comment]
Looks ok to me.

**Review by websi — CHANGES_REQUESTED** [review]
Please Update Vagrant. 
Please update gemspec

See https://github.com/dkdeploy/dkdeploy-core/compare/v8.0.0...v9.0.0

**Review by websi — DISMISSED** [review]

**Review by websi — CHANGES_REQUESTED** [review]
Please check travis build

**Review by websi — APPROVED** [review]

**Pull request merged**